### PR TITLE
transfer token 

### PR DESCRIPTION
**Pull request by xiaojiqing**
1. add `Allowance` in Tasks process
2. transfer tokens to data provider and compute nodes

### CODE REVIEW

File: src/base.lua [hunk 1]
```diff
@@ -1,6 +1,7 @@
 NODE_PROCESS_ID = NODE_PROCESS_ID or "Vlq4jWP6PLRo0Msjnxp8-vg9HalZv9e8tiz13OTK3gk"
 DATA_PROCESS_ID = DATA_PROCESS_ID or "daYyE-QRXg2MBrX1E1lUmJ1hMR-GEmyrdUiUnv3dWLY"
-TOKEN_PROCESS_ID = TOKEN_PROCESS_ID or "daYyE-QRXg2MBrX1E1lUmJ1hMR-GEmyrdUiUnv3dWLY"
+TOKEN_PROCESS_ID = TOKEN_PROCESS_ID or "Sa0iBLPNyJQrwpTTG-tWLQU-1QeUAJA73DdxGGiKoJc"
+TOKEN_FOR_COMPUTATION = 1
```
Comment (fksyuan) [inline src/base.lua]: TOKEN_FOR_COMPUTATION name can change to TOKEN_FOR_COMPUTATION_PRICE?

File: src/tasks.lua [hunk 2]
```diff
@@ -342,3 +336,24 @@ Handlers.add(
     replySuccess(msg, allTasks)
   end
 )
+Handlers.add(
```
Comment (fksyuan) [inline src/tasks.lua]: The method must be deleted. The completed tasks should not be deleted.

File: src/tasks.lua
Comment (fksyuan) [inline src/tasks.lua:1]: Do need to add a withdraw function? After the user submits the task, can I withdraw the recharged money immediately?

**Review by fksyuan — CHANGES_REQUESTED** [review]

File: src/tasks.lua (hunk 2 above)
Comment (xiaojiqing) [reply src/tasks.lua]: good advice

File: src/base.lua (hunk 1 above)
Comment (xiaojiqing) [reply src/base.lua]: good advice

File: src/tasks.lua
Comment (xiaojiqing) [reply src/tasks.lua:1]: i will add withdraw handler

File: js_test/tasks.ts [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+function sleep(ms: number) {
+    return new Promise((resolve) => setTimeout(resolve, ms))
+}
+
+async function main() {
+    const signer = await getSigner()
+    let address = await getWalletAddress()
+    const nodes = ["js_aos1", "js_aos2", "js_aos3"]
+
+    await testAddWhiteList(address, signer)
```
Comment (ujnss) [inline js_test/tasks.ts]: Calling this directly will return `Forbitten to operate white list`, should `setNodeRegistryManager` be called first? @xiaojiqing

**Review by fksyuan — APPROVED** [review]

**Pull request merged**